thirdeye/thirdeye-pinot/src/test/java/com/linkedin/thirdeye/tools/CleanupAndRegenerateAnomaliesTool.java

```diff
@@ -122,25 +123,11 @@ private void deleteExistingAnomalies() {
       AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
       LOG.info("Beginning cleanup of functionId {} collection {} metric {}",
           functionId, anomalyFunction.getCollection(), anomalyFunction.getMetric());
-      // Find merged anomalies and delete them first
-      List<MergedAnomalyResultDTO> mergedResults =
-          mergedResultDAO.findByStartTimeInRangeAndFunctionId(startTime, endTime, functionId);
-      if (CollectionUtils.isNotEmpty(mergedResults)) {
-        deleteMergedResults(mergedResults);
-      }
-      // Just in case unmerged raw anomalies exist and hence are not deleted in the previous step
-      List<RawAnomalyResultDTO> rawResults = rawResultDAO.findAllByTimeAndFunctionId(startTime, endTime, functionId);
-      if (CollectionUtils.isNotEmpty(rawResults)) {
-        for (int i = rawResults.size() -1; i >= 0; --i) {
-          if (!rawResults.get(i).isMerged()) {
-            rawResults.remove(i);
-          }
-        }
-        deleteRawResults(rawResults);
-      }
+
+      // Clean up merged and raw anomaly of functionID
+      OnboardResource onboardResource = new OnboardResource();
```

Comment:
When we run this tool, the global DAO registry is not initialized. Therefore, onboardResource cannot get the global registry it needs.

You could create a constructor that takes a given DAO registry and it uses that instead of the global one. The global on is used only when the default constructor is invoked.